packages/story-editor/src/app/pageCanvas/test/usePageCanvas.js

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { act, renderHook } from '@testing-library/react-hooks';
+/**
+ * Internal dependencies
+ */
+import { PageCanvasProvider, usePageCanvas } from '..';
+import storyPageToCanvas from '../../../utils/storyPageToCanvas';
+import useStory from '../../story/useStory';
+import createMockPage from '../testUtils/createMockPage';
+import { MockStoryProvider, useStoryMock } from '../testUtils/useStoryMock';
+
+jest.mock('../../story/useStory');
+jest.mock('../../../utils/storyPageToCanvas');
+
+jest.mock('../../../utils/contrastUtils', () => {
+  return {
+    __esModule: true,
+    getAccessibleTextColorsFromPixels: () => {},
+  };
+});
+jest.mock('../getPixelDataFromCanvas', () => {
+  return {
+    __esModule: true,
+    default: () => {},
+  };
+});
+
+function renderUsePageCanvas(_mockStoryContext) {
+  return renderHook(() => [usePageCanvas(), useStoryMock()], {
+    wrapper: ({ children }) => (
+      <MockStoryProvider mockContextValue={_mockStoryContext}>
+        <PageCanvasProvider>{children}</PageCanvasProvider>
+      </MockStoryProvider>
+    ),
+  });
+}
+
+describe('usePageCanvas', () => {
+  let mockStoryContext;
+  let currentPage;
+
+  beforeEach(() => {
+    storyPageToCanvas.mockImplementation(
+      () =>
+        'lets pretend this is a canvas. I wont tell the computer if you dont 🤫'
+    );
+
+    currentPage = createMockPage();
+    mockStoryContext = {
+      actions: {},
+      state: {
+        pages: [currentPage],
+        currentPage,
+        selectedElementIds: [],
+      },
+    };
+
+    useStory.mockImplementation(useStoryMock);
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe('calculateAccessibleTextColors', () => {
+    describe('page canvas cache', () => {
+      it('generates a page canvas when called and canvas cache is not populated', async () => {
+        const { result } = renderUsePageCanvas(mockStoryContext);
+        const element = { id: 'a' };
+
+        // cache populates with generated canvas for currentPage
+        await act(async () => {
+          const [usePageCanvasCurrent] = result.current;
+          await usePageCanvasCurrent.actions.calculateAccessibleTextColors(
+            element
+          );
+        });
+        expect(storyPageToCanvas).toHaveBeenCalledTimes(1);
+      });
```

Comment:
💭 should this also check to see if the canvas cache is not populated at the beginning?

Reply:
sure we can add it in there